AppCenterDataStorage/AppCenterDataStorage/MSSerialization.h

```diff
@@ -0,0 +1,27 @@
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol MSSerialization
+
+/**
+ * Create a dictionary from the object.
+ *
+ * @return Dictionary representing the object.
+ */
+@required
+- (NSDictionary *)serializeToDictionary;
```

Comment:
I had a similar [comment](https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1378#discussion_r265812247) about the `serializeToDictionary` naming in a different PR:
> serialization by definition means transform a data structure into a format (like string) that can be transmitted.

What about `convertToDictionary`?

Reply:
this method name is used  in many places in our code, e.g.
```@implementation MSCommonSchemaLog
- (NSMutableDictionary *)serializeToDictionary { }```

I kept it consistent with the rest. but your point is valid.
I don't think the current name is confusing to user though